include/deal.II/base/tensor_function_parser.h

```diff
@@ -0,0 +1,361 @@
+ * most right one).
+ *
+ * @note In contrast to the FunctionParser class the TensorFunctionParser class does not support
+ * 	 automatic differentiation.
+ *
+ * A minimal example for the usage of the class would be:
+ * @code
+ * // set up time dependent tensor function:
+ * std::string variables = "x,y,t";
+ * std::string expression = "exp(-t)*cos(x+y);-sin(pi*x*y-t);sin(pi*x*y-t);exp(t)*cos(x+y)";
+ * std::map<std::string,double> constants;
+ * constants["pi"] = numbers::PI;
+ *
+ * // TensorFunctionParser with 2+1 variables (space + time) in 2D of rank 2:
+ * TensorFunctionParser<2,2> tfp;
+ * tfp.initialize(variables,
+ *               expression,
+ *               constants,
+ *               true);       // This tells the parser that
+ *                            // it is a time-dependent function
+ *                            // and there is another variable
+ *                            // to be taken into account (t).
+ *
+ * // Point at which we want to evaluate the function
+ * Point<2> point(0.0, 1.0);
+ *
+ * // evaluate the expression at 'point':
+ * double result = tfp.value(point);
+ *
+ * deallog << "Function '" << expression << "'"
+ *         << " @ " << point
+ *         << " is "
+ *         << std::endl
+ *         << result[0][0] << " " << result[0][1]
+ *         << result[1][0] << " " << result[1][1]
+ *         << std::endl;
+ * @endcode
+ *
+ * See also the documentation of the FunctionParser class.
+ *
+ * @note This class implements the list of functions just mentioned as
+ *   user-defined functions by extending muparser. This means, in particular,
+ *   that the `if(condition, then-value, else-value)` syntax evaluates all
+ *   three arguments before determining whether the condition is true, and
+ *   then discarding either the "then" or the "else" expressions. In almost
+ *   all situations, this is not a problem except if the evaluation of
+ *   one of the expressions throws a floating point exception in cases
+ *   where it will later be discarded. (Assuming floating point exceptions
+ *   are switched on, as is the default for deal.II in debug mode on most
+ *   systems.) An example would be the expression `if(x>0, sqrt(x), 0)`
+ *   which is mathematically well defined, but on systems where this is
+ *   enabled will abort the program with a floating point exception when
+ *   evaluated with a negative `x`. This is because the square root of
+ *   `x` is computed before the `if` statement's condition is considered
+ *   to determine whether the result should be the second or third
+ *   argument. If this kind of behavior is a problem, you can resort to
+ *   the muparser built-in syntax `(condition ? then-value : else-value)`,
+ *   using the ternary syntax familiar to C++ programmers. If this
+ *   syntax is used, muparser uses lazy evaluation in which only one of the
+ *   branches is evaluated, depending on whether the `condition` is
+ *   true or false.
```

Comment:
I think it would be better to refer to the documentation of `FunctionParser` instead of copying and pasting this block.

Reply:
I removed the note completely